### ISSUE TITLE
Error when creating new attribute

### ISSUE THREAD

**Issue by tjpugh** [pughlab/tracker#174]
I get this error when creating a new attribute "DNA volume uL" for the REACT study:

"Sorry! Unexpected error from service handling PUT /api/studies/REACT/schema: undefined"

**Comment by morungos** [comment]
Hmm, and ooh!! That should not happen.I'm sure there'll be a more detailed error message on the server for me to look at.

**Comment by tjpugh** [comment]
Samah has this error as well when she tried to add tpugh@ads.uhnresearch.ca to a user group. In fact, she gets this error for every change made to KOMBAT tracker. Sadly, it looks like all of her data gets wiped out when she logs out. Possibly related?

**Comment by morungos** [comment]
Unlikely to be the same, unless something like the database connections have all crapped out. Let me see if I can find my one-time pad and dial in for a look see

**Comment by morungos** [comment]
OK, here's the corresponding backtrace:

```
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Column 'type' cannot be null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:411)
	at com.mysql.jdbc.Util.getInstance(Util.java:386)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1039)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3597)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3529)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:1990)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2151)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2625)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:2119)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2415)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2333)
	at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2318)
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeUpdate(DelegatingPreparedStatement.java:105)
	at com.mysema.query.sql.dml.SQLInsertClause.execute(SQLInsertClause.java:380)
	at ca.uhnresearch.pughlab.tracker.dao.impl.StudyRepositoryImpl$7.doInSqlInsertClause(StudyRepositoryImpl.java:306)
	at org.springframework.data.jdbc.query.QueryDslJdbcTemplate$8.doInConnection(QueryDslJdbcTemplate.java:263)
```

**Comment by morungos** [comment]
It looks like this was a missing type.

**Comment by morungos** [comment]
And indeed, that replicates it nicely. The dropdown isn't filled in by default, and we're not doing enough validation so people (i.e., Trevor) can save when they shouldn't ought to be able to. Of course, the problem is that the save button is in another scope.

**Comment by morungos** [comment]
See also #181

**Comment by morungos** [comment]
Should be resolved in 71d873b
